tools/shared/src/main/scala/org/scalajs/core/tools/linker/checker/IRChecker.scala

```diff
@@ -255,19 +268,25 @@ private final class IRChecker(unit: LinkingUnit, logger: Logger) {
   }
 
   private def checkExportedMethodDef(methodDef: MethodDef,
-      classDef: LinkedClass): Unit = withPerMethodState {
+      classDef: LinkedClass, isTopLevel: Boolean): Unit = withPerMethodState {
     val MethodDef(static, StringLiteral(name), params, resultType, body) = methodDef
     implicit val ctx = ErrorContext(methodDef)
 
-    if (!classDef.kind.isAnyScalaJSDefinedClass) {
+    if (isTopLevel && classDef.kind != ClassKind.ModuleClass) {
+      reportError(s"TopLevel export can only appear in a module")
+      return
```

Comment:
While I understand why top-level methods can only appear in Scala `object`s at the source level, I don't think there is any reason to forbid them anywhere else at the IR level.

In fact, we are almost obliged to allow them in Scala `class`es, at the very least, because our linker can turn objects into classes if they're not instantiated. Consider this snippet of code:
```scala
object Foo {
  @ExportTopLevel("bar")
  def bar(): Int = 42
}
```
In the IR, we'll get something like:
```scala
module class Foo$ {
  def bar__I(): int = 42
  export top static def "bar"(): any = mod:Foo$.bar__I()
}
```
Now suppose the optimizer processes `"bar"` (which it doesn't, but it could). It would inline `Foo.bar`, and throw away the module load:
```scala
module class Foo$ {
  def bar__I(): int = 42
  export top static def "bar"(): any = 42
}
```
Susquently, the Refiner would turn `Foo$` into a (non-module) class, because it is not instantiated anymore:
```
class Foo$ {
  export top static def "bar"(): any = 42
}
```
That shouldn't freak out the IR checker nor the emitter.